### PR TITLE
Add requireIsAnnotatable for better errors when annotating literals

### PR DESCRIPTION
**Pull request by jackkoenig**
### Contributor Checklist

- [x] Did you add Scaladoc to every public function/method?
- [x] Did you add at least one test demonstrating the PR?
- [x] Did you delete any extraneous printlns/debugging code?
- [x] Did you specify the type of improvement?
- [ ] Did you add appropriate documentation in `docs/src`?
- [x] Did you request a desired merge strategy?
- [x] Did you add text to be included in the Release Notes for this change?

#### Type of Improvement

- Feature (or new API)

#### Desired Merge Strategy

- Squash

#### Release Notes

This gives much better error messages when accidentally dontTouching a literal

### Reviewer Checklist (only modified by reviewer)
- [ ] Did you add the appropriate labels? (Select the most appropriate one based on the "Type of Improvement")
- [ ] Did you mark the proper milestone (Bug fix: `3.6.x`, `5.x`, or `6.x` depending on impact, API modification or big change: `7.0`)?
- [ ] Did you review?
- [ ] Did you check whether all relevant Contributor checkboxes have been checked?
- [ ] Did you do one of the following when ready to merge:
  - [ ] Squash: You/ the contributor `Enable auto-merge (squash)`, clean up the commit message, and label with `Please Merge`.
  - [ ] Merge: Ensure that contributor has cleaned up their commit history, then merge with `Create a merge commit`.

### CODE REVIEW

File: core/src/main/scala/chisel3/dontTouch.scala [hunk 1]
```diff
@@ -2,10 +2,11 @@
 
 package chisel3
 
-import chisel3.experimental.{annotate, requireIsHardware, ChiselAnnotation}
+import chisel3.experimental.{annotate, requireIsAnnotatable, ChiselAnnotation}
 import chisel3.properties.Property
 import chisel3.reflect.DataMirror
 import firrtl.transforms.DontTouchAnnotation
+import chisel3.experimental.requireIsHardware
```
Comment (seldridge) [inline core/src/main/scala/chisel3/dontTouch.scala:9]: It seems like this should be combined in the first import.

**Review by seldridge — APPROVED** [review]
LGTM

File: core/src/main/scala/chisel3/dontTouch.scala (hunk 1 above)
Comment (jackkoenig) [reply core/src/main/scala/chisel3/dontTouch.scala:9]: That second one is actually unused, I readded it by accident (for testing), will remove!

Comment (jackkoenig) [reply core/src/main/scala/chisel3/dontTouch.scala]: Fixed!

**Pull request merged**